src/rpc.rs

```diff
@@ -79,7 +81,7 @@ pub async fn send_matched_orders(
 
     let payload: MatchRequest = MatchRequest { maker, taker };
     let client: Client = Client::new();
-    let endpoint: String = address + "/submit";
+    let endpoint: String = address.clone() + "/submit";
```

Comment:
Should we do the same with the check orders function above? I know that it is cloned on input but maybe safer to clone within the function?

Reply:
This is purely due to borrowing semantics later on in the function. Given neither function mutate the values being cloned I don't see why? Clippy will likely complain on needless cloning also.